nixos/modules/tasks/network-interfaces-systemd.nix

```diff
@@ -83,8 +83,11 @@ in
           name = mkDefault i.name;
           DHCP = mkForce (dhcpStr
             (if i.useDHCP != null then i.useDHCP else cfg.useDHCP && interfaceIps i == [ ]));
-          address = flip map (interfaceIps i)
-            (ip: "${ip.address}/${toString ip.prefixLength}");
+          addresses = flip map (interfaceIps i)
+            (ip: { addressConfig = {
+              Address = "${ip.address}/${toString ip.prefixLength}";
+              Broadcast = mkIf (ip ? "broadcast" && ip.broadcast != null) ip.broadcast;
```

Comment:
Here there is a check if  `broadcast` exists.

Reply:
Yeah, it's needed to distinguish IPv6 entries from IPv4 entries, similarly to the one in the scripted implementation that fpletz references below.